File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/common/EncoderUtils.java
package com.zancheema.android.pantry.common;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

public class EncoderUtils {
    public static String getBasicAuthHeader(String username, String password) {
        String valueToEncode = username + ":" + password;
        String encodedValue = Base64
                .getEncoder()
                .encodeToString(valueToEncode.getBytes(StandardCharsets.UTF_8));
        return "Basic " + encodedValue;
    }
}

File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/common/EncoderUtils.kt
package com.zancheema.android.pantry.common

import java.nio.charset.StandardCharsets
import java.util.*

object EncoderUtils {
    fun getBasicAuthHeader(username: String, password: String): String {
        val valueToEncode = "$username:$password"
        val encodedValue = Base64
            .getEncoder()
            .encodeToString(valueToEncode.toByteArray(StandardCharsets.UTF_8))
        return "Basic $encodedValue";
    }
}